source/common/common/assert.h

```diff
@@ -126,6 +126,32 @@ bool shouldLogAndInvokeEnvoyBugForEnvoyBugMacroUseOnly(absl::string_view bug_nam
  */
 void resetEnvoyBugCountersForTest();
 
+/**
+ * Sets an action to be invoked when an ENVOY_NOTIFICATION is encountered.
+ *
+ * This function is not thread-safe; concurrent calls to set the action are not allowed.
+ *
+ * The action may be invoked concurrently if two ENVOY_NOTIFICATION in different threads run at the
+ * same time, so the action must be thread-safe.
+ *
+ * The action will be invoked in all build types (debug or release).
+ *
+ * @param action The action to take when an envoy bug fails.
+ * @return A registration object. The registration is removed when the object is destructed.
+ */
+ActionRegistrationPtr
+addEnvoyNotificationRecordAction(const std::function<void(absl::string_view)>& action);
```

Comment:
Should these be moved to a different namespace? Code-wise they're related to `Assert`, but functionally they're a bit different.

Reply:
Done